resources/todomvc/big-dom-generator/utils/buildComplex.js

```diff
@@ -2,32 +2,51 @@ const fs = require("fs");
 const { JSDOM } = require("jsdom");
 const path = require("path");
 
-const TARGET_DIRECTORY = "./dist";
-const COMPLEX_DOM_HTML_FILE = "index.html";
-const TODO_HTML_FILE = "index.html";
-const CSS_FILES_TO_ADD_LINKS_FOR = ["big-dom-generator.css"];
+function buildComplex(options) {
+    const {
+        callerDirectory,
+        sourceDirectory,
+        title,
+        filesToMove,
+        cssFilePath,
+        cssFolder = "", // sometimes the css file we are looking for may be nested in another folder.
+        cssFileNamePattern, // The css file name pattern is used to find the css file in the source dist directory.
+        extraCssToLink = [],
+        scriptsToLink = [],
+        targetDirectory = "./dist",
+        complexDomHtmlFile = "index.html",
+        todoHtmlFile = "index.html",
+        cssFilesToAddLinksFor = ["big-dom-generator.css"],
+    } = options;
 
-function buildComplex(CALLER_DIRECTORY, SOURCE_DIRECTORY, TITLE, FILES_TO_MOVE, EXTRA_CSS_TO_LINK = [], SCRIPTS_TO_LINK = []) {
     // remove dist directory if it exists
-    fs.rmSync(path.resolve(TARGET_DIRECTORY), { recursive: true, force: true });
+    fs.rmSync(path.resolve(targetDirectory), { recursive: true, force: true });
 
     // re-create the directory.
-    fs.mkdirSync(path.resolve(TARGET_DIRECTORY));
+    fs.mkdirSync(path.resolve(targetDirectory));
 
     // copy dist folder from javascript-es6-webpack
-    fs.cpSync(path.join(CALLER_DIRECTORY, SOURCE_DIRECTORY), path.resolve(TARGET_DIRECTORY), { recursive: true });
+    fs.cpSync(path.join(callerDirectory, sourceDirectory), path.resolve(targetDirectory), { recursive: true });
 
     // copy files to move
-    for (let i = 0; i < FILES_TO_MOVE.length; i++) {
+    for (let i = 0; i < filesToMove.length; i++) {
         // rename app.css to big-dom-generator.css so it's unique.
-        const sourcePath = path.resolve(CALLER_DIRECTORY, "..", FILES_TO_MOVE[i]);
-        const fileName = path.basename(FILES_TO_MOVE[i]);
-        const targetPath = path.join(TARGET_DIRECTORY, fileName);
+        const sourcePath = path.resolve(callerDirectory, "..", filesToMove[i]);
+        const fileName = path.basename(filesToMove[i]);
+        const targetPath = path.join(targetDirectory, fileName);
         fs.copyFileSync(sourcePath, targetPath);
     }
 
+    if (cssFilePath) {
+        // get the name of the css file that's in the dist, we do this because the name of the css file may change
+        const cssFile = fs.readdirSync(path.join(callerDirectory, sourceDirectory, cssFolder), { withFileTypes: true }).find((dirent) => dirent.isFile() && cssFileNamePattern.test(dirent.name))?.name;
```

Comment:
I can see that this project is ignored by eslint and prettier (see .eslintignore and .prettierignore), it would probably be good to not ignore the handcrafted files such as this one and the others such as the handcrafted CSS file.

Reply:
I'm good if you want to do this in a follow-up to more clearly distinguish the concerns.

Reply:
IIUC this project is not ignored, both the rules added to the .prettierignore and .eslintignore both start with "!".

I also see these files in the output when I run the formatter.
![image](https://github.com/WebKit/Speedometer/assets/117320405/30e430a3-e3fa-46cd-bab5-4f614e1bc090)

I can remove the rules I added previously in a follow-up since they aren't needed and I already have a follow-up PR in the works.
```
!/resources/todomvc/big-dom-generator
```

Reply:
oh right I got confused.
Our project's prettier is configured with `printWidth: 250`. I find this too long and got confused by this long line, but looks like this is accepted by prettier then...
Maybe you could split it up a bit still, up to you.